go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -2835,6 +2856,230 @@ func (s *VtctldServer) ValidateKeyspace(ctx context.Context, req *vtctldatapb.Va
 	return &resp, nil
 }
 
+// ValidateSchemaKeyspace is a part of the vtctlservicepb.VtctldServer interface.
+// It will diff the schema from all the tablets in the keyspace.
+func (s *VtctldServer) ValidateSchemaKeyspace(ctx context.Context, req *vtctldatapb.ValidateSchemaKeyspaceRequest) (*vtctldatapb.ValidateSchemaKeyspaceResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.ValidateSchemaKeyspace")
+	defer span.Finish()
+
+	span.Annotate("keyspace", req.Keyspace)
+	keyspace := req.Keyspace
+
+	resp := vtctldatapb.ValidateSchemaKeyspaceResponse{}
+
+	shards, err := s.ts.GetShardNames(ctx, keyspace)
+	if err != nil {
+		resp.Results = append(resp.Results, fmt.Sprintf("TopologyServer.GetShardNames(%v) failed: %v", req.Keyspace, err))
+		return &resp, nil
+	}
+
+	resp.ResultsByShard = make(map[string]*vtctldatapb.ValidateShardResponse, len(shards))
+
+	sort.Strings(shards)
+
+	var (
+		referenceSchema *tabletmanagerdatapb.SchemaDefinition
+		referenceAlias  *topodatapb.TabletAlias
+	)
+
+	var (
+		m  sync.Mutex
+		wg sync.WaitGroup
+	)
+
+	// TO DO: Use `IncludeVschema` parameter to determine whether or not to also add results from ValidateVschema to the results array.
+	// Unimplemented for now because ValidateVSchema method is out of scope.
```

Comment:
do we have an issue filed for this? if not, can you make one?

Reply:
No issue - I'll file one!

Reply:
https://github.com/vitessio/vitess/issues/9732 added one here 🙂